python/gink/impl/database.py

```diff
@@ -268,31 +277,68 @@ def run(self,
             until = self.resolve_timestamp(until)
         context_manager = console or nullcontext()
         with context_manager:
-            while (until is None or generate_timestamp() < until):
-                # eventually will want to support epoll on platforms where its supported
-                readers: List[Union[Listener, Connection, SelectableConsole, AbstractStore]] = []
-                if self._store.is_selectable():
+            # eventually will want to support epoll on platforms where its supported
+            readers: List[Union[Listener, Connection, SelectableConsole, AbstractStore, Socket]] = []
+            if self._store.is_selectable():
                     readers.append(self._store)
-                for listener in self._listeners:
-                    readers.append(listener)
-                for connection in list(self._connections):
-                    if connection.is_closed():
-                        self._connections.remove(connection)
-                    else:
-                        readers.append(connection)
-                if isinstance(console, SelectableConsole):
-                    readers.append(console)
+            for listener in self._listeners:
+                readers.append(listener)
+            for connection in list(self._connections):
+                if connection.is_closed():
+                    self._connections.remove(connection)
+                else:
+                    readers.append(connection)
+            if self._wsgi_server:
+                readers.append(self._wsgi_server.listen_socket)
+            if isinstance(console, SelectableConsole):
+                readers.append(console)
+            while (until is None or generate_timestamp() < until):
```

Comment:
Every time the select loop started again, the readers were reinitialized. I'm not sure if that was intentional, but the way I have it here works much better for the web server and I think it is faster regardless

Reply:
The reason for rebuilding the readers list is that it can change over time.  E.g. if a connection is closed then it needs to be removed from the list of things you're selecting on.  You can also add to the set of listeners from commands executed via the console.